### PR TITLE
Wallet address validation

### PR DESCRIPTION
**Pull request by jamierumbelow**
Validates wallet addresses according to the EIP-55 checksum, or the upper/lowercase status otherwise. Also a few sanity checks (is it 42 chars? does it begin with `0x`? etc)

UI could do with some tweaking (as could the code.) But, in the interests of shipping...

Also allows wallet names to be optional, and makes a few changes to facilitate this.

### CODE REVIEW

File: Balance/Table Cells/WalletTableViewCell.swift [hunk 1]
```diff
@@ -37,13 +29,13 @@ class WalletTableViewCell: UITableViewCell {
         return label
     }()
     
-    private let containerView: UIView = {
+    internal let containerView: UIView = {
```
Comment (einsteinx2) [inline Balance/Table Cells/WalletTableViewCell.swift]: `internal` is not correct here. It essentially does nothing when used outside of a framework as it acts the same as `public` or no modifier at all, `private` is the correct modifier as these are meant to be visible only to this class. Internal only makes sense when you want something to be public, but only inside of a framework.

File: Balance/Table Cells/WalletTableViewCell.swift [hunk 2]
```diff
@@ -37,13 +29,13 @@ class WalletTableViewCell: UITableViewCell {
...
         let view = UIView()
         view.backgroundColor = .clear
         return view
     }()
     
-    private let backgroundImageView: UIImageView = {
+    internal let backgroundImageView: UIImageView = {
```
Comment (einsteinx2) [inline Balance/Table Cells/WalletTableViewCell.swift]: `internal` is not correct here. It essentially does nothing when used outside of a framework as it acts the same as `public` or no modifier at all, `private` is the correct modifier as these are meant to be visible only to this class. Internal only makes sense when you want something to be public, but only inside of a framework.

**Review by einsteinx2 — CHANGES_REQUESTED** [review]

File: Balance/Table Cells/WalletTableViewCell.swift (hunk 1 above)
Comment (jamierumbelow) [reply Balance/Table Cells/WalletTableViewCell.swift]: Does `private` allow overriding by subclasses? `renderTableViewCellContentFor`, `nameLabel` and others are set to `internal` for this reason.

Comment (einsteinx2) [reply Balance/Table Cells/WalletTableViewCell.swift]: Ah sorry missed that you were subclassing. In that case just remove the modifier. The default is `internal` so it's superfluous.

Comment (jamierumbelow) [reply Balance/Table Cells/WalletTableViewCell.swift]: Changed to `private` since I'm not overriding these properties anyway :)

**Review by einsteinx2 — APPROVED** [review]

**Pull request merged**